### DISCUSSION TITLE
Camera `to_world` differentiability

### DISCUSSION THREAD

**Discussion by satyajitghana** [mitsuba-renderer/mitsuba3#374] [General]
Hi, is there any example to get camera `to_world` to be differentiable?

I've been trying, but didn't work: https://github.com/mitsuba-renderer/mitsuba3/pull/251

**Comment by njroussel** [comment]
Hi @satyajitghana,

The PR you linked adds a test case in `src/integrators/tests/test_ad_integrators.py` which is a setup where the sensor is being translated. 
You can use that as an example.

The output you posted in the PR is rather surprising. 
>  (...)/pytorch3d/lib/python3.9/site-packages/mitsuba/python/ad/optimizers.py:51, in Optimizer.__setitem__(self, key, value)

Did you manually install your own build? If not, I think your current setup is picking up an old `mitsuba` install which doesn't have the PR yet.

Also, in the future, could you please just stick to a single thread - spreading this conversation makes it hard to follow for us or anyone that would have the same issue :)

**Reply by satyajitghana** [reply]
Yes, I've built the latest build.

I'll look into the test case. I tried to understand that, but it was a little difficult. Thanks though.
